### DISCUSSION TITLE
Wireguard VPN Connect OK, Internet OK, No ping devices on Server network

### DISCUSSION THREAD

**Discussion by gibsonmb** [pivpn/pivpn#1820] [Support] [answered]
OK, I've installed pivpn server on a Pi4B running Bookworm. Installation went fine. I've installed Wireguard vpn clients on an android phone, a laptop and my brothers PC (he is in the UK, I am in France). All clients can connect to the pivpn server fine. All clients can browse the internet using Chrome over their respective vpn connections. However we want this vpn so we can connect to devices on the same network as the pi vpn server. Issue, we cannot ping any of the devices (including the gateway router 192.168.1.1) on the same network as the pi hosting the pivpn server. I've pasted the pivpn server config below. Any ideas/help please folks?

PLAT=Debian
OSCN=bookworm
USING_UFW=0
IPv4dev=eth0
IPv6dev=eth0
dhcpReserv=1
IPv4addr=192.168.1.42/24
IPv4gw=192.168.1.1
install_user=pi
install_home=/home/pi
VPN=wireguard
pivpnPORT=51820
pivpnDNS1=208.67.222.222
pivpnDNS2=208.67.220.220
pivpnHOST=mydomainname
INPUT_CHAIN_EDITED=0
FORWARD_CHAIN_EDITED=0
INPUT_CHAIN_EDITEDv6=0
FORWARD_CHAIN_EDITEDv6=0
pivpnPROTO=udp
pivpnMTU=1420
pivpnDEV=wg0
pivpnNET=10.47.224.0
subnetClass=24
pivpnenableipv6=1
pivpnNETv6="fd11:5ee:bad:c0de::"
subnetClassv6=64
ALLOWED_IPS="0.0.0.0/0, ::0/0"
UNATTUPG=1
INSTALLED_PACKAGES=(dnsutils grepcidr bsdmainutils iptables-persistent wireguard-tools qrencode unattended-upgrades)

**Comment by orazioedoardo** [comment]
Likely due to using 192.168.1.0/24 on both ends.

**Comment by jerryfudd** [comment]
Did you get anywhere with this? after working fine for 6 months on Wireguard this week mine has decided to stop working.

Connects fine, but no routing to internal addresses as you describe

**Reply by gibsonmb** [reply, marked as answer]
I did, in my case it was overlapping subnets at each end of the vpn

**Reply by jerryfudd** [reply]
how did you resolve that? ...exactly

**Reply by gibsonmb** [reply]
In my case I was testing with my brother from a remote location. We happened to have the same sub-net. The sub-net at my own home was different. Home is 192.168.0.xxx remote location is 192.168.1.xxx so no issue in the operational scenario. So the way to fix this is just to use different sub-nets on each end of the vpn. There may be more technically complex solutions but I didn't need to investigate further.

**Reply by jerryfudd** [reply]
oh I see, yeah that would be an issue - had that before and had to do a static route for a specific address I was trying to hit

back to the drawing board on my issue :D ....going to give OpenVPN again, was using that for ages before WireGuard

thanks!
